Tasks/Gulp/gulptask.js

```diff
@@ -1,24 +1,34 @@
 var path = require('path');
 var tl = require('vso-task-lib');
 
-var gt = new tl.ToolRunner(tl.which('gulp', true));
+var nt = new tl.ToolRunner(tl.which('node', true));
 
-// optional - no tasks will concat nothing
-gt.arg(tl.getDelimitedInput('targets', ' ', false));
-
-gt.arg('--gulpfile');
 var gulpFile = tl.getPathInput('gulpFile', true);
-gt.arg(gulpFile);
-
-gt.arg(tl.getDelimitedInput('arguments', ' ', false));
-
 var cwd = tl.getInput('cwd', false);
 if (!cwd) {
 	cwd = path.dirname(gulpFile);
 }
+
+var gulpjs = tl.getInput('gulpjs', false, true);
+if (!gulpjs) {
+	gulpjs = path.resolve(path.join(cwd, 'node_modules/gulp/bin/gulp.js'));
```

Comment:
See my note about it being required.
That allows you to do 
var gulpjs = path.resolve(cwd, tl.getInput('gulpjs', false, true)

path.resolve is the equivalent of two CDs in a row.  So, if the second is fully qualified /usr/local/bin/foo/... then it will win.  If it's relative then they are combined.  it's what you want - the input is 'applied' against cwd.  So, imagine you are in the cwd and do CD <input>.  That's the path you get.